### ISSUE TITLE
Output messages

### ISSUE THREAD

**Issue by mihaifireball** [fventuri/navtex#2]
it is possible to output each NAVTEX message in separate file? 
Thank you!

**Comment by fventuri** [comment]
@mihaifireball - not with the code as it is now.

However I just took a quick look, and I saw that the file handle `m_messagesfile` is only used in one place in `navtex_rx.cpp` (https://github.com/fventuri/navtex/blob/main/src/navtex_rx.cpp#L386), so perhaps the easiest way to achieve what you want would be to replace that line (and the line above) with something like this:
```
    // some code here to generate the file name based for instance on a pattern
    m_messagesfile = fopen(filename, "w");
    if (m_messagesfile != nullptr) {
        fputs(message.c_str(), m_messagesfile);
        fclose(m_messagesfile);
    }
```

**Comment by mihaifireball** [comment]
Thanks!
It is working, but the header (ex. ZCZC PA41) and the end of message (NNN) is lost.
Also, I would like the name of the name of the file to include the header (ex. PA41) and current time and date.
This is how it looks like now:
`

     // some code here to generate the file name based for instance on a pattern
    char filename[40];
    struct tm *timenow;
    time_t now = time(NULL);
    timenow = gmtime(&now);
    strftime(filename, sizeof(filename), "XXXX_%Y-%m-%d_%H:%M:%S.txt", timenow);    
    m_messagesfile = fopen(filename, "w");
    if (m_messagesfile != nullptr) {
        fputs(message.c_str(), m_messagesfile);
        fclose(m_messagesfile);
    }
`
I want to replace XXXX in the filename with header (PA41).

Also, the output message has extra "new line":
`
      
     NETHERLANDS COASTGUARD

     NAVIGATIONAL WARNING NR. 19 212040 UTC MAY

     WINDFARM BORSSELE

      GEOPHYSICAL SURVEY BY BIBBY TETHRA 2EGF8

      A. 51-41N 003-08E

      B. 51-41N 002-58E

      C. 51-49N 003-08E

      D. 51-49N 002-58E

      WIDE BERTH REQUESTED

`

**Comment by fventuri** [comment]
@mihaifireball - to have not just the NAVTEX message body but also the header and the `NNN` trailer, I think you have to put a similar logic in the `process_messages` function instead  (I would say somewhere around this line: https://github.com/fventuri/navtex/blob/main/src/navtex_rx.cpp#L885).

As per the extra new line you are seeing it is possibly due to the fact that, if I remember correctly, each line in those NAVTEX messages ends with carriage return + line feed ('\r\n'). You can get rid of the last character before the call to `fputs()` with something like this:
```
    message[strlen(message)-1] = '\0';
```

Franco

**Comment by mihaifireball** [comment]
I have tried, but without any success on those issues... It's out of my understanding.. 🙂
Thank you for your support!
You may close this issue. 
73!

**Comment by fventuri** [comment]
@mihaifireball
Sorry to hear that; this afternoon I thought that perhaps it is easier to go back to the original code from the repository, have it write to stdout, and then pipe it into a simple Python script to split that stream into different files like you want.

Tonight I found a few minutes to write that Python script; I ran it though a test file I found and it seems to work.
The script is attached here; if you know a little Python it should be easy for you to change it in case you find problems.

Franco

[parse_navtex.zip](https://github.com/fventuri/navtex/files/10108957/parse_navtex.zip)
